### PR TITLE
Task 11 - GET /api/articles query refactor

### PR DESCRIPTION
**Pull request by clam119**
For this PR I have done the following:

1. Add sort_by & order queries to the getAllArticles function in the articles controller. 
2. Add conditional logic to the fetchAllArticles models function that will by default provide a sort_by query of "created_at" and a default order query of "DESC" if none are provided by the user. However, if a user does provide any of these queries but they are not acceptable, then a 400 error will be returned.
3. Add tests for successful queries that return a 200 status code and with the correct order & sort_by criteria as well as 400 error code tests for invalid queries given by the user.

Probably the most confusing task yet to refactor with so much logic that I'd need to look to go over but the tests are passing for now 😭

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -266,6 +266,53 @@ describe("Northcoders News API", () => {
...
+          const articleValues = Object.values(article);
+          expect(articleValues.includes('mitch')).toBe(true);
+        })
+        expect(mitchArticles).toBeSortedBy("votes", {descending: true});
+      })
+    })
+
+    it("Status: 400 - Should respond with invalid data type message if passed in an invalid sort_by query", () => {
+      return request(app)
+      .get("/api/articles?sort_by=TIMETOSQLINJECT")
```
Comment (alexdevdavis) [inline __tests__/app.test.js:300]: haha, great query value

**Review by alexdevdavis — APPROVED** [review]
Agreed, this needs some refactoring, as it's a massive solution, but it works!
Consideration:
- how might you cut out a whole bunch of code by setting default sort and order values in your model.

File: __tests__/app.test.js (hunk 1 above)
Comment (clam119) [reply __tests__/app.test.js:300]: Hahaha I'm glad you like it Alex🤣

**Comment by clam119** [comment]
> Agreed, this needs some refactoring, as it's a massive solution, but it works! Consideration:
> 
> * how might you cut out a whole bunch of code by setting default sort and order values in your model.

Would I perhaps just go into the controller and have a conditional saying that if they're undefined, then just assign them a default value before they're passed into fetchAllArticles as the parameters & invoked?

**Pull request merged**